src/object/tcb.c

```diff
@@ -1604,9 +1642,15 @@ exception_t decodeDomainInvocation(word_t invLabel, word_t length, word_t *buffe
         current_syscall_error.invalidArgumentNumber = 1;
         return EXCEPTION_SYSCALL_ERROR;
     }
+    tcb = TCB_PTR(cap_thread_cap_get_capTCBPtr(tcap));
+    return invokeDomainSetSet(tcb, domain);
```

Comment:
Oh and for completeness, now that the two uses of `tcb` have been pulled into a separate function it's up to you whether you prefer it as a separate variable like this or inline in the return like it was before. Either option now shouldn't impact the proofs.